pkg/metallb/frr.go

```diff
@@ -0,0 +1,583 @@
+func NewFrrConfigurationBuilder(
+	apiClient *clients.Settings, name, nsname string) *FrrConfigurationBuilder {
+	glog.V(100).Infof(
+		"Initializing new Frrconfiguration structure with the following params: %s, %s",
+		name, nsname)
+
+	err := apiClient.AttachScheme(frrtypes.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add metallb scheme to client schemes")
+
+		return nil
+	}
+
+	builder := FrrConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &frrtypes.FRRConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the frrConfiguration is empty")
+
+		builder.errorMsg = "frrConfiguration 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the frrConfiguration is empty")
+
+		builder.errorMsg = "frrConfiguration 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Create makes a FrrConfiguration in the cluster and stores the created object in struct.
+func (builder *FrrConfigurationBuilder) Create() (*FrrConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the frrconfiguration %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+
+		if err != nil {
+			glog.V(100).Infof("Failed to create MetalLb")
+
+			return nil, err
+		}
+
+		builder.Object = builder.Definition
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return builder, err
+}
+
+// Get returns FRRConfiguration object if found.
+func (builder *FrrConfigurationBuilder) Get() (*frrtypes.FRRConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof(
+		"Collecting FRRConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	frrConfig := &frrtypes.FRRConfiguration{}
+	err := builder.apiClient.Get(context.TODO(),
+		runtimeClient.ObjectKey{Name: builder.Definition.Name, Namespace: builder.Definition.Namespace}, frrConfig)
+
+	if err != nil {
+		glog.V(100).Infof(
+			"metallb object %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		return nil, err
+	}
+
+	return frrConfig, nil
+}
+
+// Delete removes FrrConfigurationBuilder object from a cluster.
+func (builder *FrrConfigurationBuilder) Delete() (*FrrConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the FrrConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("FrrConfiguration cannot be deleted because it does not exist")
+	}
```

Comment:
we're slowly moving towards delete not returning an error if it does not exist, like in [ocm/kac.go](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/ocm/kac.go#L245-L253)

Reply:
changed